### PR TITLE
Adding manifest generation steps

### PR DESCRIPTION
**Pull request by marcschier**
* This change adds the SBOM manifest generation to the CI build (ci.yaml).
* It also updates the operation limits in the server to make integration tests run more smoothly. (ServerFactory.cs)
* Remove the use of username password in the release/import script.
* Update publisher documentation.  Remove marketplace snapshot (out of date) and qualify hosting environment.
* Finally some additional cleanup.

### CODE REVIEW

File: docs/modules/publisher.md [hunk 1]
```diff
@@ -10,8 +10,6 @@ It runs on [Azure IoT Edge](https://azure.microsoft.com/en-us/services/iot-edge/
 
 Please use our released containers for OPC Publisher available in the Microsoft Container Registry, rather than building from sources. The easiest way to deploy OPC Publisher is through the [Azure Marketplace](https://azuremarketplace.microsoft.com/en-us/marketplace/apps/microsoft_iot.iotedge-opc-publisher).
 
-[<img src="../media/image-20201028141833399.png" style="zoom:50%;" />](https://azuremarketplace.microsoft.com/en-us/marketplace/apps/microsoft_iot.iotedge-opc-publisher)
```
Comment (karok2m) [inline docs/modules/publisher.md:13]: @marcschier, the image does not seem to show up in documentation on your branch. Can you please have a look ?

image

File: tools/templates/ci.yml [hunk 2]
```diff
@@ -50,12 +50,34 @@ jobs:
       command: build
       projects: '$(file)'
       arguments: '--configuration Release'
+  - task: DotNetCoreCLI@2
+    inputs:
```
Comment (karok2m) [inline tools/templates/ci.yml:55]: Please add a descriptive `displayName` to the task. It shows up as `DotNetCoreCLI` now.

File: tools/templates/ci.yml [hunk 3]
```diff
@@ -50,12 +50,34 @@ jobs:
...
+  - publish: $(System.ArtifactsDirectory)/$(agent)_Release
+    artifact: $(agent)_Release
   - task: DotNetCoreCLI@2
     displayName: Debug Build
     inputs:
       command: build
       projects: '$(file)'
       arguments: '--configuration Debug'
+  - task: DotNetCoreCLI@2
+    inputs:
```
Comment (karok2m) [inline tools/templates/ci.yml:74]: Please add a descriptive `displayName` to the task. It shows up as `DotNetCoreCLI` now.

File: docs/modules/publisher.md (hunk 1 above)
Comment (karok2m) [reply docs/modules/publisher.md:13]: Sorry, marked the wrong place with the arrow. It should be between first and second lines under `Getting Started`.

**Review by karok2m — APPROVED** [review]

**Review by cristipogacean — APPROVED** [review]

**Pull request merged**